src/ripple/conditions/PrefixSha256.h

```diff
@@ -0,0 +1,214 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2016 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_CONDITIONS_PREFIX_SHA256_H
+#define RIPPLE_CONDITIONS_PREFIX_SHA256_H
+
+#include <ripple/basics/base_uint.h>
+#include <ripple/conditions/Condition.h>
+#include <ripple/conditions/Fulfillment.h>
+#include <ripple/conditions/impl/base64.h>
+#include <ripple/protocol/digest.h>
+#include <boost/regex.hpp>
+#include <boost/optional.hpp>
+#include <algorithm>
+#include <iterator>
+#include <memory>
+#include <vector>
+
+namespace ripple {
+namespace cryptoconditions {
+
+class PrefixSha256 final
+    : public Fulfillment
+{
+    Buffer prefix_;
+    std::unique_ptr<Fulfillment> subfulfillment_;
+
+public:
+    PrefixSha256 () = default;
+
+    std::size_t
+    payloadSize () const override
+    {
+        return subfulfillment_->payloadSize();
+    }
+
+    Buffer
+    payload() const override
+    {
+        auto subpayload = to_blob (*subfulfillment_);
```

Comment:
`subfulfillment_` will be empty on default construction or on a parse error. Since this function is called from `operator==`, we need to handle an empty `subfulfillment_` here.